mona_sdk/client.py

```diff
@@ -632,7 +664,80 @@ def get_config_history(self, number_of_revisions=UNPROVIDED_VALUE):
             "get_config_history", data={"number_of_revisions": number_of_revisions}
         )
 
+    def _update_sampling_factors_if_needed(self):
+        """
+        If the client was initiated with a sampling config name, check if the
+        configuration was changed since the client vars were assigned, and if so, update
+        them accordingly.
+        """
+        if self._sampling_config_name:
+            # Refetch the updated config from the index (if the response is not cached).
+            sampling_config = self.get_sampling_factors()[0]
+            default_from_index = sampling_config.get("default_factor")
+            factors_map_from_index = sampling_config.get("factors_map")
+
+            if (
```

Comment:
Most of the invocations to this function won't change the config so we should have a lean mechanism in this case, and we do more work only if something had changed. So I suggest to store the latest sampling_config in a class variable (for example self._latest_seen_sampling_config). The constructor will store the result of get_sampling_factors into it. Here you first check if sampling_config equals to it, and if so return right away. Otherwise start extracting, update the individual params and do proper logging.